app/lib/frontend/name_tracker.dart

```diff
@@ -71,14 +71,18 @@ class NameTracker {
 class NameTrackerUpdater {
```

Comment:
Shameless self-promotion, have you considered: https://pub.dev/packages/neat_periodic_task

If you ignore the status provider part...it has both start and stop.. and it has a timeout :)
 + there are some tests that start/stop works, and of what happens when errors occur... + a bit of logging..

Just saying :)

Note: this is not important.